c2/c2-protocol/c2-protocol-api/pom.xml

```
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>c2-protocol</artifactId>
+        <groupId>org.apache.nifi</groupId>
+        <version>1.16.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>c2-protocol-api</artifactId>
```

Should this be prefixed with `nifi`?
```suggestion
    <artifactId>nifi-c2-protocol-api</artifactId>
```

---

Since we have `nifi` in the group name and the C2 modules are meant to be used by NiFi but not strictly tied to it, I'd like to keep it just `c2` like we do with `minifi`, unless there are strong objections?

---

Thanks for the reply. For clarification, this is intended to support not just MiNiFi, but be more generic across the board?  MiNiFi modules seem to be the exception when it comes to artifact name prefixing, and other components, such as nifi-commons, include the `nifi` prefix.  Although `nifi` is included in the group name, that is true across the board.  When viewing the generated JAR artifacts, the group name is not present, and in this case, having the `nifi` prefix helps indicate that this is a C2 protocol for NiFi.  If we had other examples outside of MiNiFi, but inside the repository, without the prefix, then it would seem less necessary, but as it stands, I lean more in the direction of keeping the prefix.

To clarify, you are seeing that this C2 module is not necessarily specific to NiFi at all?  That may be a stronger argument for leaving it as-is, but just trying to understand the scope a bit more.

---

I envision this module being used by anyone that wants a straightforward solution to C2, so I'd like to keep it not specific to NiFi and keep the name without the prefix to strengthen that point.

---

Thanks for the additional background, I agree, that sounds good.